x-pack/plugins/ml/public/url_generator.ts

```diff
@@ -65,40 +163,214 @@ export class MlUrlGenerator implements UrlGeneratorsDefinition<typeof ML_APP_URL
   public readonly id = ML_APP_URL_GENERATOR;
 
   public readonly createUrl = async ({ page, ...params }: MlUrlGeneratorState): Promise<string> => {
-    if (page === 'explorer') {
-      return this.createExplorerUrl(params);
+    if (page === ML_TABS.ANOMALY_DETECTION) {
```

Comment:
Please remove destructuring from the function arguments, and change to something like 
```suggestion
public readonly createUrl = async (mlUrlGeneratorState: MlUrlGeneratorState): Promise<string> => {
    switch(mlUrlGeneratorState) {
      case mlUrlGeneratorState.page === ML_TABS.ANOMALY_DETECTION;
      return this.createAnomalyDetectionJobManagementUrl(mlUrlGeneratorState)
    }
```
in that case, TS will take care of typecasting and there is no need to do it manually.

Reply:
This is a good spot. And with this way it's much cleaner too! Thanks 🙏